packages/compass-connections/src/hooks/use-connection-repository.ts

```diff
@@ -145,7 +168,7 @@ export function useConnectionRepository(): ConnectionRepository {
         infoToSave.connectionOptions?.connectionString
       );
 
-      if (!persistOIDCTokens) {
+      if (!preferences.getPreferences().persistOIDCTokens) {
```

Comment:
Drive-by to make the function more stable, we only care about the current value, we don't really need to update the whole callback for this, accessor works better here